contracts/src/token_locker.cairo

```diff
@@ -0,0 +1,177 @@
+        locker: ContractAddress,
+        token: ContractAddress,
+        lock_time: u64,
+        amount: u256
+    }
+
+    #[derive(Drop, starknet::Event)]
+    struct TokenUnlocked {
+        unlocker: ContractAddress,
+        token: ContractAddress,
+        unlock_time: u64,
+        amount: u256
+    }
+
+    #[storage]
+    struct Storage {
+        lock_time: u64,
+        locks: LegacyMap<(ContractAddress, ContractAddress, u64), u256>,
+    }
+
+    /// Constructor called once when the contract is deployed.
+    /// # Arguments
+    /// * `lock_time` - Locking period as timestamp.
+    #[constructor]
+    fn constructor(ref self: ContractState, lock_time: u64,) {
+        self.lock_time.write(lock_time);
+    }
+
+    #[abi(embed_v0)]
+    impl TokenLocker of super::ITokenLocker<ContractState> {
+        /// Locks tokens from callers account
+        /// # Arguments
+        /// * `token_contract` - Address of token contract that is going to be locked
+        /// * `amount` - Amount of tokens that is going to be locked
+        fn lock(ref self: ContractState, token_contract: ContractAddress, amount: u256) {
+            assert(amount != 0, ZERO_LOCK);
+
+            let caller = get_caller_address();
+            let current_time = get_block_timestamp();
+            assert(self.locks.read((token_contract, caller, current_time)) == 0, LOCK_EXIST);
+
+            self.locks.write((token_contract, caller, current_time), amount);
+
+            let this_address = get_contract_address();
+            let initial_balance = ERC20ABIDispatcher { contract_address: token_contract }
+                .balanceOf(this_address);
+            ERC20ABIDispatcher { contract_address: token_contract }
+                .transferFrom(caller, this_address, amount);
+            assert(
+                ERC20ABIDispatcher { contract_address: token_contract }.balanceOf(this_address)
+                    - initial_balance == amount,
+                TRANSFER_FAIL
+            );
+
+            self
+                .emit(
+                    TokenLocked {
+                        locker: caller,
+                        token: token_contract,
+                        lock_time: current_time,
+                        amount: amount
+                    }
+                );
+        }
+
+        /// Unlocks tokens to callers account
+        /// # Arguments
+        /// * `token_contract` - Address of token contract that is going to be unlocked
+        /// * `lock_time` - Initial lock timestamp
+        fn unlock(ref self: ContractState, token_contract: ContractAddress, lock_time: u64) {
+            assert(lock_time != 0, ZERO_LOCK_TIME);
+
+            let caller = get_caller_address();
+            let current_time = get_block_timestamp();
+            let locked_amount = self.locks.read((token_contract, caller, lock_time));
+
+            assert(locked_amount != 0, LOCK_NONEXIST);
+            let lock_end_time = lock_time + self.lock_time.read();
```

Comment:
the confusion is very visible here: lock_time + self.lock_time. No errors here but this is typically how bugs happen

Reply:
okay after rename i guess its also okay